src/context/cart-context.js

```diff
@@ -0,0 +1,185 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { cartReducer } from "../reducer/cartReducer";
+import { useAuth } from "../context/auth-context";
+
+const CartContext = createContext();
+
+const CartProvider = ({ children }) => {
+  const navigate = useNavigate();
+  const { loggedIn } = useAuth();
+
+  const initialCartState = {
```

Comment:
You can move this object outside of the component body. It will be recreated every time the component re rebders. You can avoid this by moving it outside the component body